components/inputs/input-date.js

```diff
@@ -212,12 +222,35 @@ class InputDate extends LocalizeStaticMixin(LitElement) {
 		super.updated(changedProperties);
 
 		changedProperties.forEach((oldVal, prop) => {
-			if (prop === 'value') {
-				this._formattedValue = this.value ? formatISODateInUserCalDescriptor(this.value) : '';
+			if (prop === '_dateTimeDescriptor' || prop === 'value') {
+				this._setFormattedValue();
 			}
 		});
 	}
 
+	_getContentWidth() {
+		const text = document.createElement('div');
+		document.body.appendChild(text);
+		text.style.fontSize = '0.8rem';
+		text.style.width = 'auto';
+		text.style.position = 'absolute';
+
+		// in some languages (e.g., fr) placeholderWidth is bigger, in others (e.g., zh) contentWidth is bigger
+		text.innerHTML = (this._dateTimeDescriptor.formats.dateFormats.short).toUpperCase();
```

Comment:
If we can use `innerText` instead of `innerHTML` it avoids possible HTML-injection attacks, which I'd especially be worried about coming from `emptyStateText`.